pysindy/optimizers/stlsq.py

```diff
@@ -157,6 +188,18 @@ def _no_change(self):
             last_coef = np.zeros_like(this_coef)
         return all(bool(i) == bool(j) for i, j in zip(this_coef, last_coef))
 
+    def _remove_and_decrement(
+        self, existing_vals: np.ndarray, vals_to_remove: np.ndarray
+    ) -> np.ndarray:
+        """Remove the sparse indices that have been thresholded
+        for subsequent iterations"""
```

Comment:
Edit the docstring to the context of the function - nothing here about sparsity, thresholding, iterations, or even that `existing_vals` stores indices.